app/views/admin/AdministerUserModal.coffee

```diff
@@ -247,4 +260,164 @@ module.exports = class AdministerUserModal extends ModalView
           @prepaidTableState[prepaidId] = 'viewMode'
           @renderSelectors('#'+prepaidId)
         return
- 
\ No newline at end of file
+
+  userIsSchoolAdmin: -> @user.isSchoolAdmin()
+
+  onClickSchoolAdminCheckbox: (e) ->
+    checked = @$(e.target).prop('checked')
+    @userSaveState = 'saving'
+    @render()
+    fetchJson("/db/user/#{@user.id}/schoolAdministrator", {
+      method: 'PUT',
+      json: {
+        schoolAdministrator: checked
+      }
+    }).then (res) =>
+      @userSaveState = 'saved'
+      @render()
+      setTimeout((()=>
+        @userSaveState = null
+        @render()
+      ), 2000)
+    null
+
+  onClickEditSchoolAdmins: (e) ->
+    if typeof @editingSchoolAdmins is 'undefined'
+      administrated = @user.get('administratedTeachers')
+
+      if administrated?.length
+        $.ajax
```

Comment:
Should we add this to an API method in `app/core/api`?

Reply:
@jmif Oops, I didn't see this when it was first posted. Is this comment still relevant? 

If so, do you mean having the https://github.com/codecombat/codecombat-server/pull/59 endpoint move away from `/db/user`?

Reply:
Yup I think its still relevant.  We're going to need this call in two places (here and in the school admin dashboard itself) and the frontend has a collection of API methods in `app/core/api` that are basically wrappers around `fetch` calls.  I think it'd be helpful to place this call there so we can reuse.